### PR TITLE
Add plugin to integrate senkyoshi into canvas

### PR DESCRIPTION
**Pull request by bfcoder**
Update readme
Add hound.yml

This depends on [this PR](https://github.com/atomicjolt/senkyoshi/pull/56) being merged.

The Senkyoshi worker is mostly copy and paste the canvas cartridge worker code.
* I broke it into method calls to shorten the complexity of the perform.
* Added calls to Senkyoshi to convert the data and upload scorm content.

### CODE REVIEW

File: lib/senkyoshi_canvas_plugin/engine.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module SenkyoshiCanvasPlugin
+  NAME = "Senkyoshi Conversion Importer".freeze
+  DISPLAY_NAME = "Blackboard Cartridge".freeze
+  DESCRIPTION =
+    "This enables importing Blackboard Cartridges into Canvas.".freeze
+  class Engine < ::Rails::Engine
+    config.autoload_paths << File.expand_path(File.join(__FILE__, "../.."))
+
+    # initializer "senkyoshi_importer.canvas_plugin" do
```
Comment (bfcoder) [inline lib/senkyoshi_canvas_plugin/engine.rb]: I'm not sure which way is the best to register this plugin.
```ruby
initializer "senkyoshi_importer.canvas_plugin" do
```
or
```ruby
config.to_prepare do
```

File: README.md [hunk 2]
```diff
@@ -1 +1,48 @@
-# senkyoshi_canvas_plugin
+# Senkyoshi Canvas Plugin
+
+This plugin enables the use of the Senkyoshi converter inside of Canvas. It adds a blackboard cartridge option to the import content dropdown for a course.
+
+## Installation
+
+```sh
+sysadmin@appserver:~$ cd /path/to/canvas/gems/plugins
+sysadmin@appserver:/path/to/canvas/gems/plugins$ git clone https://github.com/atomicjolt/senkyoshi_canvas_plugin.git
```
Comment (blunckr) [inline README.md]: This can't just be added a gem in a gemfile?

File: lib/canvas/migration/worker/senkyoshi_worker.rb [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+ARGUMENT_ERROR =
+  "converter_class required for content migration with no file".freeze
+
+module Canvas
+  module Migration
+    module Worker
+      class SenkyoshiWorker < Base
+        def perform
+          cm ||= ContentMigration.where(id: migration_id).first
```
Comment (blunckr) [inline lib/canvas/migration/worker/senkyoshi_worker.rb]: more descriptive name? If you're further down the file it's not clear what this is.

File: lib/canvas/migration/worker/senkyoshi_worker.rb [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+          settings = cm.migration_settings.clone
+          settings[:content_migration_id] = migration_id
+          settings[:user_id] = cm.user_id
+          settings[:content_migration] = cm
+
+          if cm.attachment
+            settings[:attachment_id] = cm.attachment.id
+            settings[:file_path] = cm.attachment.full_filename
+          elsif settings[:file_url]
+            att = Canvas::Migration::Worker.
```
Comment (blunckr) [inline lib/canvas/migration/worker/senkyoshi_worker.rb]: This could be a more descriptive name as well

File: README.md (hunk 2 above)
Comment (blunckr) [inline README.md]: That would probably make it easier to install in the cluster setup. just one less step

Comment (bfcoder) [inline README.md]: I suppose I could make it a full gem that you just add a file to the Gemfile.d folder listing it. I'll do that.

**Pull request merged**